### ISSUE TITLE
CMake build support [Feature]

### ISSUE THREAD

**Issue by fklassen** [appneta/tcpreplay#688]
**Problem to fix**
Upgrade to modern build CMake tools to support advanced features for Visual Studio Code.

**Comment by GabrielGanne** [comment]
Hi,
I have a working WIP branch https://github.com/GabrielGanne/tcpreplay/tree/a7af611e7df904443cfb8a2228a60b954af90280 (branch name is `meson`) that adds meson support.
The current status is that I believe It should support all build options, but I haven't tested across different platforms (macos-sdk), nor options like netmap yet. Also to note: it depends on PR #991 .
Please tell me if you're interested, I can clean it up and open a formal PR.
Best regards,

**Comment by fklassen** [comment]
> Hi, I have a working WIP branch https://github.com/GabrielGanne/tcpreplay/tree/a7af611e7df904443cfb8a2228a60b954af90280 (branch name is `meson`) that adds meson support. The current status is that I believe It should support all build options, but I haven't tested across different platforms (macos-sdk), nor options like netmap yet. Also to note: it depends on PR [#991](https://github.com/appneta/tcpreplay/pull/991) . Please tell me if you're interested, I can clean it up and open a formal PR. Best regards,

That's great! I just created `v4.6.0-beta1` which you can target your work towards.

Unfortunately I messed up your #991 PR by merging and removing it. You can cherry-pick a new PR for 4.6, but we have to make sure that `sudo make test` works without modifications on an out-of-tree build (e.g. a build directory). That means that we also have `--load-opts` working and to restore `test/config`. Also, everything in `--more-help` needs to be displayed, however that option is not necessary.

I still would like to have CMAKE support (this ticket). I have 24 hours of cheap Claude Code Fable 5 access, so I will use up my tokens trying to get that done. 

Feel free to contact me directly to discuss.

**Comment by fklassen** [comment]
CMake build support merged in #1018 (branch v4.6.0-beta1). Every configure.ac feature has a CMake equivalent; autotools remains the canonical release build. Verified on macOS and Linux (incl. netmap); CI now builds both via the cmake-build job.
